src/puppetlabs/trapperkeeper/services/status/status_service.clj

```diff
@@ -8,7 +8,11 @@
 (defprotocol StatusService
   (register-status [this service-name service-version status-version status-fn]
     "Register a status callback function for a service by adding it to the
-    status service context."))
+    status service context.  status-fn must be a function of arity 1 which takes
+    the status level as a keyword and returns the status information for
+    the given level.  The return value of the callback function must satisfy
+    the puppetlabs.trapperkeeper.services.status.status-core/StatusCallbackResponse
```

Comment:
Right, I guess it's not strictly enforced, but status functions still need to conform to it (i.e. `:is-running` should most likely be there, and the rest of the data needs to be under `:status` for it to end up in the response).  Perhaps I could have explained that a bit better in this docstring.